### PR TITLE
Rename LossConfig.loss to LossConfig.instance in fv3fit

### PR DESCRIPTION
**Pull request by mcgibbon**
This PR makes small fixes to fv3fit.

Refactored public API:
- fv3fit.pytorch's LossConfig.loss is renamed to LossConfig.instance
- pytest configuration now ignores certain deprecation warnings internal to tensorflow/pytorch

Significant internal changes:
- Adjusted type hinting on `fv3fit.tfdataset.apply_to_mapping` and `fv3fit.tfdataset.apply_to_tuple` so they can be used for arbitrary structures of tensors

No new tests needed.

Coverage reports (updated automatically):
- test_unit: [83%](https://output.circle-artifacts.com/output/job/f1b32e89-32f6-43b7-921b-81306b412dce/artifacts/0/tmp/coverage/htmlcov-test_unit/index.html)

### CODE REVIEW

**Review by brianhenn — APPROVED** [review]
LGTM
